### PR TITLE
Fix publication pages bugs

### LINKED ISSUE #2545: publications navigation bug
- Navigating to pages in the publications gives the correct c0ontent
- If you go directly to the URL, the context is incorrect

The two pages below have the same URL and different content.

Screenshot from 2022-11-17 16-14-41

- [ ] ensure that the same URL has the same content, regardless of how you get there.

### LINKED ISSUE #2547: internal links don't work in publications static pages
There are internal links in publications pages, namely:

- ToC
- endnotes

The HTML is well-formed, but for some reason the links do not go to the right target.

http://localhost/edition/snp/en/sujato/introduction

### PR DESCRIPTION
**Pull request by ihongda**

### CODE REVIEW

**Comment by ihongda** [comment]
1. Fix publication matter page navigation bug.
2. Optimize code structure.
3. Fix some small bugs.

**Comment by sujato** [comment]
Okay, I am looking at this branch now.

- [x] go to editions page
- [x] download all editions in all types
- [x] go to github source

Great!

Now go to "long Discourses" page

- [ ] page needs to be restructured: it uses the old format, the correct format is on staging
- [ ] Hardcover book should be removed, it is not available (this is correct on staging)
- [ ] paperback printing will be added soon
- [x] EPUB: click and download
- [x] PDF: click and download
- [x] Raw HTML: this doesn't download, it just opens a raw HTML page. It should download a zip file. Leave it for now.
- [x] Raw TeX: downloads a TeX file
- [x] Web: goes to the right link

Pretty good, we just need to remove the hardcover and add the links for the printing.

Other details:

- [ ] the images are still incorrect, they use the dummy images I made long ago. Use the appropriate EPUB cover for each thumbnail.
- [ ] the image text (`alt` and `<figcaption>` is incorrect, it must be dynamic text for each edition, currently it uses the same text everywhere.
- [ ] on the main "Editions" page, remove the Vinaya edition: this is not ready yet.

**Comment by ihongda** [comment]
> the image text (alt and <figcaption> is incorrect, it must be dynamic text for each edition, currently it uses the same text everywhere.

Bhante, Apart from the author `uid`, such as `sujato`, currently no additional information can be displayed.

**Comment by sujato** [comment]
> > the image text (`alt` and `figcaption` is incorrect, it must be dynamic text for each edition, currently it uses the same text everywhere.
> 
> Bhante, Apart from the author `uid`, such as `sujato`, currently no additional information can be displayed.

Sorry, I don't understand? Can we not add the title?

**Review by sujato — APPROVED** [review]

File: client/elements/publication/sc-publication-common.js [hunk 1]
```diff
@@ -1,13 +1,75 @@
+import { API_ROOT } from '../../constants';
+import { reduxActions } from '../addons/sc-redux-actions';
+import { store } from '../../redux-store';
```
Comment (agilgur5) [inline client/elements/publication/sc-publication-common.js:3]: `reduxActions` and `store` are unused imports here -- they were added in 457658359d9663c452fbf3f386cb48ce0fa20c0b, but the code that used it moved outside of it in 1cba1691af43be7f966b211cdba0908e4ecd5b15. I've removed these imports in #3727

**Pull request merged**